File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count=0
        temp=head
        while(temp!=None):
            temp=temp.next
            count+=1
        target=count-(n-1)
        index=1
        temp=head
        prev=None
        while(temp!=None):
            if(index==target):
                if prev is None:
                    head=head.next
                    break
                else:
                    prev.next=temp.next
                    break
            prev=temp
            temp=temp.next
            index+=1
        return head

File: Data Structures & Algorithms/palindromic-substrings/submission-0.java
class Solution {
    public int countSubstrings(String s) {
        int n=s.length();
        Boolean [][]dp= new Boolean[n][n];
        int count=0;
        for(int i=0;i<n;i++){
            for(int j=i;j<n;j++){
                if(solver(i,j,s,dp)){
                    count++;
                }
            }
        }
        return count;
    }

    private Boolean solver(int i,int j,String s,Boolean [][]dp){
        if(i>=j){
            return true;
        }
        if(dp[i][j]!=null){
            return dp[i][j];
        }
        if(s.charAt(i)!=s.charAt(j)){
            dp[i][j]=false;
            return dp[i][j];
        }
        dp[i][j]=solver(i+1,j-1,s,dp);
        return dp[i][j];
    }
}
